apply/processor/utils.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"github.com/sealerio/sealer/pkg/runtime"
+	"github.com/sealerio/sealer/pkg/runtime/k0s"
+	"github.com/sealerio/sealer/pkg/runtime/kubernetes"
+	"github.com/sealerio/sealer/pkg/runtime/kubernetes/kubeadm"
+	v2 "github.com/sealerio/sealer/types/api/v2"
+)
+
+func RuntimeChoose(rootfs string, cluster *v2.Cluster, config *kubeadm.KubeadmConfig) (runtime.Interface, error) {
```

Comment:
@starComingup Here I have a point to discuss with you. 

If runtime is a field of details processor instance, or belongs to a processor's function?

If runtime is a field of processor, then RuntimeChoose should be used in a processor's initialization, rather than set runtime in processor's function.

For example, I see 
```
type ScaleProcessor struct {
    runtime runtime.Interface
    xxx
}
```

And I see 
```
func (s *ScaleProcessor) PreProcess(cluster *v2.Cluster) error {
	runTime, err := RuntimeChoose(platform.DefaultMountClusterImageDir(cluster.Name), cluster, s.ClusterFile.GetKubeadmConfig())
	if err != nil {
		return fmt.Errorf("failed to init default runtime: %v", err)
	}
	s.Runtime = runTime
```
I feel quite strange. Why do you `RuntimeChoose` in ScaleProcessor and set ScaleProcessor's runtime filed, instead of set runtime type in `NewScaleProcessor`? Initialization in NewScaleProcessor could make code much more clear and more object-oriented. 

WDYT?

Reply:
In addition, please change the function name into `ChooseRuntime` or `GetRuntime` rather than `RuntimeChoose`.

Reply:
> Here I have a point to discuss with you.
> 
> If runtime is a field of details processor instance, or belongs to a processor's function?
> 
> If runtime is a field of processor, then RuntimeChoose should be used in a processor's initialization, rather than set runtime in processor's function.
> 
> For example, I see

The old code set kubernetes.runtime here, so I just make this. But when I coded this, I also felt confused why origin logic set runtime in sub function. I will try to modify it!